src/test/java/seedu/address/model/components/name/NameContainsKeywordsPredicateTest.java

```diff
@@ -0,0 +1,87 @@
+package seedu.address.model.components.name;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+
+public class NameContainsKeywordsPredicateTest {
+
+    @Test
+    public void equals() {
+        List<String> firstPredicateKeywordList = Collections.singletonList("first");
+        List<String> secondPredicateKeywordList = Arrays.asList("first", "second");
+
+        NameContainsKeywordsPredicate<HasNameStub> firstPredicate =
+                new NameContainsKeywordsPredicate<>(firstPredicateKeywordList);
+        NameContainsKeywordsPredicate<HasNameStub> secondPredicate =
+                new NameContainsKeywordsPredicate<>(secondPredicateKeywordList);
+
+        // same object -> returns true
+        assertTrue(firstPredicate.equals(firstPredicate));
+
+        // same values -> returns true
+        NameContainsKeywordsPredicate<HasNameStub> firstPredicateCopy =
+                new NameContainsKeywordsPredicate<>(firstPredicateKeywordList);
+        assertTrue(firstPredicate.equals(firstPredicateCopy));
+
+        // different types -> returns false
+        assertFalse(firstPredicate.equals(1));
+
+        // null -> returns false
+        assertFalse(firstPredicate.equals(null));
+
+        // different module class -> returns false
```

Comment:
Perhaps this should change to `different objects` ?
```suggestion
        // different objects -> returns false
```